provider-directory/src/main/java/gov/va/api/health/providerdirectory/service/controller/practitioner/PractitionerController.java

```diff
@@ -148,13 +138,111 @@ private PractitionerWrapper search(MultiValueMap<String, String> parameters) {
   public Practitioner.Bundle searchByName(
       @RequestParam("name") String name,
       @RequestParam(value = "page", defaultValue = "1") @Min(1) int page,
-      @RequestParam(value = "_count", defaultValue = "1") @Min(0) int count) {
+      @CountParameter @Min(0) int count) {
     return bundle(
         Parameters.builder().add("name", name).add("page", page).add("_count", count).build(),
         page,
         count);
   }
 
+  /** Logic for search by Family and Given. */
+  private Pair<List<PractitionerWrapper>, Integer> searchFamilyGiven(
+      MultiValueMap<String, String> parameters) {
+    String familyName = parameters.getFirst("family");
+    String givenName = parameters.getFirst("given");
+    ProviderResponse providerResponse = ppmsClient.providersForName(familyName);
+    int page = Integer.parseInt(parameters.getOrDefault("page", singletonList("1")).get(0));
+    int count = Integer.parseInt(parameters.getOrDefault("_count", singletonList("15")).get(0));
+    int fromIndex = Math.min((page - 1) * count, providerResponse.value().size());
+    int toIndex = Math.min((fromIndex + count), providerResponse.value().size());
+    /** Retrieve a list of providerResponse from PPMS using familyName. */
+    List<ProviderResponse.Value> providerResponseUnfilteredPages =
+        providerResponse.value().subList(fromIndex, toIndex);
+    /** Remove any providerResponse that doesn't contain the givenName. */
+    List<ProviderResponse.Value> providerResponsePages = new ArrayList<>();
+    for (int i = 0; i < providerResponseUnfilteredPages.size(); i++) {
+      if (StringUtils.containsIgnoreCase(
+          providerResponseUnfilteredPages.get(i).name(), givenName)) {
+        providerResponsePages.add(providerResponse.value().get(i));
+      }
+    }
+    /** Using providerResponse, retrieve a list of providerContactsResponse from PPMS. */
+    List<ProviderContactsResponse> providerContactsResponsePages =
+        providerResponsePages
+            .parallelStream()
+            .map(
```

Comment:
I think this can be simplified a little...

Something like 
.map(prv -> ppmsClient.providerContactsForId(prv.providerIdentifier().toString())).collect(Collectors.toList());